<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1047</width>
    <height>635</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="RenderWidget" name="renderWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>6</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="focusPolicy">
        <enum>Qt::StrongFocus</enum>
       </property>
      </widget>
      <widget class="QFrame" name="controls">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>2</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="baseSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="4" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Bend Dampening</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSlider" name="dampingStiffnessSlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Bend Stiffness</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSlider" name="stretchStiffnessSlider">
            <property name="maximum">
             <number>5000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSlider" name="bendDampeningSlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Mass</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSlider" name="massSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Dampening</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>Stiffness</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSlider" name="bendStiffnessSlider">
            <property name="maximum">
             <number>5000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSlider" name="gravitySlider">
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>Gravity</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
       <zorder></zorder>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1047</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSimulation">
    <property name="title">
     <string>Simulation</string>
    </property>
    <addaction name="actionTogglePlay"/>
    <addaction name="actionStep"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionStretchForces"/>
    <addaction name="actionBendForces"/>
    <addaction name="actionToggleForces"/>
   </widget>
   <addaction name="menuSimulation"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionTogglePlay">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="text">
    <string>Step</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionStretchForces">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Stretch Forces</string>
   </property>
   <property name="toolTip">
    <string>Stretch forces</string>
   </property>
  </action>
  <action name="actionBendForces">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Bend Forces</string>
   </property>
   <property name="toolTip">
    <string>Show bend forces</string>
   </property>
  </action>
  <action name="actionToggleForces">
   <property name="text">
    <string>Toggle Forces</string>
   </property>
   <property name="toolTip">
    <string>Toggle all forces</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RenderWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>RenderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionTogglePlay</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>TogglePlay(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>379</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stretchStiffnessSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>813</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>486</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bendStiffnessSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>SliderChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>813</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>486</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStep</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>Step()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStretchForces</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>ShowStretchForces(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBendForces</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>ShowBendForces(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleForces</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>ToggleForces()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>TogglePlay(bool)</slot>
  <slot>SliderChanged(int)</slot>
  <slot>Step()</slot>
  <slot>ShowStretchForces(bool)</slot>
  <slot>ShowBendForces(bool)</slot>
  <slot>ToggleForces()</slot>
 </slots>
</ui>
